cthulhu/cthulhu/manager/cache_pool_request_factory.py

```diff
@@ -0,0 +1,57 @@
+from cthulhu.manager.request_factory import RequestFactory
+from cthulhu.manager.user_request import OsdMapModifyingRequest
+from cthulhu.log import log
+from calamari_common.types import OsdMap
+
+
+class CachePoolRequestFactory(RequestFactory):
+
+    def create(self, attributes):
+        commands = [('osd tier add', {"pool": attributes['pool'], "tierpool": attributes['tier_pool']})]
+        if 'mode' in attributes and attributes['mode'] is not None:
+            commands.append(('osd tier cache-mode', {'pool': attributes['tier_pool'], 'mode': attributes['mode']}))
+            if attributes['mode'] == 'write_back':
+                commands.append(
+                    ('osd tier set-overlay', {"pool": attributes['pool'], "overlaypool": attributes['tier_pool']}))
+        else:
+            log.warn("there is not cache-mode")
+        message = "Creating cache pool {cache_pool_name} for pool {pool_name}".format(
+            pool_name=attributes['pool'], cache_pool_name=attributes['tier_pool'])
+        return OsdMapModifyingRequest(message, self._cluster_monitor.fsid, self._cluster_monitor.name, commands)
+
+    def delete(self, attributes):
+        pool_id = attributes['pool']
+        tier_pool_id = attributes['tier_pool']
+        tier_name = self._resolve_pool(tier_pool_id)['pool_name']
+        pool_name = self._resolve_pool(pool_id)['pool_name']
+        cache_mode = self._resolve_pool(pool_id)['cache_mode']
+        commands = []
+        if cache_mode == "write_back":
```

Comment:
@GregMeno  And here except writeback mode the read-forward and read-proxy also need to do the same operation drain the cache pool before turning it off.